Feature/ban unsafe imgs

abrzezinski94 opened the pull request.

asktree commented on components/NewRealmWizard/components/TokenInput.tsx:
```diff
@@ -82,13 +83,9 @@ export default function TokenInput({
-      setTokenList(filteredTokenList)
+      const tokens = await axios.get('https://token.jup.ag/strict')
+      const tokenList = tokens.data as TokenInfo[]
+      setTokenList(tokenList)
```
let's also use react-query for this because we would really prefer to cache it

asktree commented on hooks/queries/tokenMetadata.ts:
```diff
@@ -0,0 +1,40 @@
+
+  const enabled = !!mint && !!otherConditions
+
+  const query = useQuery({
```
is token metadata even mutable generally? 

we can do `useQuery({cacheTime: (10 days in miliseconds), staleTime: (1hr in miliseconds)` for example, and then it will just get cached a long time and not get refreshed more than each hour

asktree commented on components/TreasuryAccount/AccountItem.tsx:
```diff
@@ -19,63 +17,19 @@ const AccountItem = ({
+    !logo
+  )
 
   useEffect(() => {
```
just use `useMemo` here instead of a state + effect

asktree commented on components/TreasuryAccount/AccountsTabs.tsx:
```diff
@@ -65,42 +63,20 @@ const AccountTab: FunctionComponent<AccountTabProps> = ({
-    }
-    if (!logo && assetAccount.extensions.mint?.publicKey.toBase58()) {
-      getTokenMetadata(assetAccount.extensions.mint?.publicKey.toBase58())
+  useEffect(() => {
```
same, just use `useMemo` instead of state + effect.

asktree commented on components/TreasuryAccount/BaseAccountHeader.tsx:
```diff
@@ -16,39 +14,20 @@ const BaseAccountHeader: React.FC<{
-      getTokenMetadata(mintAddress)
     }
-  })
+    if (data?.symbol) {
```
`useMemo`

asktree commented on components/NewRealmWizard/components/NFTCollectionSelector.tsx:
```diff
@@ -16,6 +16,7 @@ function ImageWithLoader({ className, ...props }) {
       </div>
       <img
         {...props}
+        src={props.src}
```
?

abrzezinski94 replied:
...props are not typed and it dont recognize there is props.src

abrzezinski94 replied:
ill add htmlInputProps

asktree commented on hooks/queries/tokenMetadata.ts:
```diff
@@ -0,0 +1,40 @@
+      )
+
+      const state = Metadata.deserialize(accountData!.data)
+      const jsonUri = state[0].data.uri.slice(
```
😮 what about just using `Metadata.fromAccountAddress` and then just `result.uri` or whatever? this looks harder to maintain in future

asktree commented on hooks/queries/tokenMetadata.ts:
```diff
@@ -0,0 +1,40 @@
+      )
+
+      const data = await (await fetch(jsonUri)).json()
+      //we dont want unsafe imgs
```
i dont follow this comment -- are imgs being filtered here?

abrzezinski94 replied:
img is inside data. will adjust comment for people to not include imgs from metadata

asktree replied:
i think it would be better if this query just returned the data in its natural state (`Metadata`) and then the components themselves were responsible for using the right source of data for the images. you could leave a comment tho. i think by default any transformation of data is bad so better to keep to a minimum. thats how i'd do it anyway.

asktree reviewed (commented):
im not sure i understand the change. it seems like the only difference is we get metadata from  jup.ag instead of the spl-token-registry thing? struggling to follow the flow of the data here

abrzezinski94 commented:
> im not sure i understand the change. it seems like the only difference is we get metadata from jup.ag instead of the spl-token-registry thing? struggling to follow the flow of the data here

i just replaced what was on front end and removed imges from metadata we can still get symbol if there is not symbol in jupiter api.

abrzezinski94 commented:
@asktree anyway at few places e.g treasury view this metdata rate limit rpc if you have a lot of token accounts the metadata request should be batched but it needs some refactor on wallet page.
